### PR TITLE
Add support for disable_controller and force_platform, use PM multi-arch file naming for the binary, add build tags in binary (date, git)

### PR DESCRIPTION
**Pull request by cdeletre**
Supported values for `disable_controller` are `true` and `false` (default)

Supported values for `force_platform` are:
- `os_unknown`
- `os_windows`
- `os_macosx`
- `os_ios`
- `os_android` (default)
- `os_linux`
- `os_psvita`
- `os_ps4`
- `os_xboxone`
- `os_tvos`
- `os_switch`

Binaries will be named `gmloadernext.aarch64` and `gmloadernext.armhf`

Build tags can be checked this way:
```bash
root@9118ae47a15f:/devel/gmloader-next# strings build/aarch64-linux-gnu/gmloader/gmloadernext.aarch64 | grep -E  "(GIT_|BUILD_)"
GIT_HASH_3aeaec44a2283c11b630758476cdef274c2a0a5a
GIT_BRANCH_master
BUILD_DATE_20241104_110957
```

### CODE REVIEW

File: gmloader/main.cpp [hunk 1]
```diff
@@ -109,6 +110,45 @@ int main(int argc, char *argv[])
...
+        }else if(strcmp(platform_ov, "os_ps4") == 0){
+            FORCE_PLATFORM = os_ps4;
+        }else if(strcmp(platform_ov, "os_xboxone") == 0){
+            FORCE_PLATFORM = os_xboxone;
+        }else if(strcmp(platform_ov, "os_tvos") == 0){
+            FORCE_PLATFORM = os_tvos;
+        }else if(strcmp(platform_ov, "os_switch") == 0){
+            FORCE_PLATFORM = os_switch;
+        }else{
+            fatal_error("Unexpected platform '%s'.\n", platform_ov);
```
Comment (JeodC) [inline gmloader/main.cpp:144]: If we force a fatal error here does it mean the application stops? Shouldn't we just default to os_unknown or fallback to os_android?

File: Makefile.gmloader [hunk 2]
```diff
@@ -50,7 +60,7 @@ GENERIC_OBJ:=\
 
 OBJ=$(patsubst build/%,build/${ARCH}/%,${GENERIC_OBJ})
 
-all: build/${ARCH}/gmloader/gmloader
+all: build/${ARCH}/gmloader/gmloadernext.${DEVICE_ARCH}
```
Comment (JeodC) [inline Makefile.gmloader:63]: Wondering if we should drop the next part and just use gmloader as Johnny originally intended this to replace the old gmloader.

File: Makefile.gmloader [hunk 3]
```diff
@@ -60,7 +70,7 @@ clean:
 
 # By repeating the generated headers on this rule, we force them to start building
 # as soon as possible, so libc_table.cpp isn't waiting forever.
-build/${ARCH}/gmloader/gmloader: ${LIBC_TABS} $(BSD_OBJ) $(OBJ)
+build/${ARCH}/gmloader/gmloadernext.${DEVICE_ARCH}: ${LIBC_TABS} $(BSD_OBJ) $(OBJ)
```
Comment (JeodC) [inline Makefile.gmloader:73]: See above.

File: gmloader/main.cpp (hunk 1 above)
Comment (cdeletre) [reply gmloader/main.cpp:144]: in the current implementation fatal_error doesn't make the application to stop. Also it fallbacks to `os_unkown`.
```bash
./gmloadernext.aarch64 -c /devel/gmloader-next/configuration.gmld
Loading config file (/devel/gmloader-next/configuration.gmld)
	save_dir = gamedata
	apk_path = my_game.apk
	show_cursor = 0
	disable_controller = 0
	force_platform = os_wind0ws
gmloader/main.cpp:144: Unexpected platform 'os_wind0ws'.
os_type = os_unknown
gmloader/main.cpp:160: Failure opening APK '/devel/gmloader-next/build/aarch64-linux-gnu/gmloader/my_game.apk': No such file
```

However, maybe using `warning()` will be appropriate.

File: Makefile.gmloader (hunk 2 above)
Comment (cdeletre) [reply Makefile.gmloader:63]: I don't mind at all removing `next`. Let's ask Johnny :)

Comment (JeodC) [reply Makefile.gmloader:63]: @JohnnyonFlame What do you think? @cdeletre is it possible to separate gamedata from savedata in this PR as well, or do you think that would warrant a separate pull request?

**Review by JanTrueno — APPROVED** [review]

**Review by JeodC — APPROVED** [review]
Good to merge but we need to figure out the audio issue.

**Pull request merged**